interactive/models.py

```diff
@@ -142,10 +146,13 @@ def send_email(sender, instance, created, **kwargs):
     if not created:
         return
 
+    uid = urlsafe_base64_encode(force_bytes(instance.pk))
```

Comment:
Just trying to understand ULIDs a bit more here, do the function calls give us a URL-safe ULID string because `instance.pk` is a ULID object rather than a string?

Reply:
Yes. The PK is a Timeflake object and `urlsafe_base64_encode` requires a bytes-like object

`>>> a.id
Timeflake('018139afb3e8455dfcf63cf808782649')
>>> force_bytes(a.id)
b'02q30K2LWCrsSWa5btizuL'
>>> urlsafe_base64_encode(a.id)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/lucy/.pyenv/versions/interactive/lib/python3.10/site-packages/django/utils/http.py", line 190, in urlsafe_base64_encode
    return base64.urlsafe_b64encode(s).rstrip(b"\n=").decode("ascii")
  File "/home/lucy/.pyenv/versions/3.10.4/lib/python3.10/base64.py", line 118, in urlsafe_b64encode
    return b64encode(s).translate(_urlsafe_encode_translation)
  File "/home/lucy/.pyenv/versions/3.10.4/lib/python3.10/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'Timeflake'
>>> urlsafe_base64_encode(force_bytes(a.id))
'MDJxMzBLMkxXQ3JzU1dhNWJ0aXp1TA'`

Reply:
I have no idea why GH is refusing for format that code. Hopefully it's almost readable now

Reply:
Needed more backticks!

That's really useful, thank you 🙂